pkg/featuregates/featuregates.go

```diff
@@ -68,9 +65,19 @@ var defaultFeatureGates = map[featuregate.Feature]featuregate.VersionedSpecs{
 	},
 }
 
-// init instantiates and sets the singleton 'FeatureGates' variable with newFeatureGates().
-func init() {
-	FeatureGates = newFeatureGates(parseProjectVersion())
+var (
+	featureGatesOnce sync.Once
+	featureGates     featuregate.MutableVersionedFeatureGate
+)
+
+// GetFeatureGates instantiates and returns the package-level singleton representing
+// the set of all feature gates and their values.
+// It contains both project-specific feature gates and standard Kubernetes logging feature gates.
+func GetFeatureGates() featuregate.MutableVersionedFeatureGate {
+	featureGatesOnce.Do(func() {
+		featureGates = newFeatureGates(parseProjectVersion())
+	})
+	return featureGates
```

Comment:
We can avoid always having to synchronize across the `Do` function once featureGates is set. 

```suggestion
func GetFeatureGates() featuregate.MutableVersionedFeatureGate {
	if featureGates == nil {
		featureGatesOnce.Do(func() {
			featureGates = newFeatureGates(parseProjectVersion())
		})
	}
	return featureGates
```